tests/test_top_level_rtseq_params.py

```diff
@@ -0,0 +1,22 @@
+from niveristand import nivs_rt_sequence
+from niveristand import NivsParam
+from niveristand import realtimesequencetools
+from niveristand.clientapi import ChannelReference, DoubleValue
+
+
+@NivsParam('p', DoubleValue(1.2), NivsParam.BY_REF)
+@nivs_rt_sequence
+def func1(p):
+    return p.value
+
+
+def test_func1_numeric():
```

Comment:
I'd like to see a few more tests, particularly:
- Passing parameters with invalid names
- Passing parameters with incorrect datatypes
- Passing parameters by value (and getting reasonable behavior)
- "Forgetting" to pass parameters that are expected
- Tests that call directly into RealTimeSequence so we can validate that API as well
(separate pull request) - Refactor the engine demo examples to pass parameters to top-level sequence instead of having a sub-top-level sequence that passes the parameters.

Reply:
i have added the tests that you have asked for and i have also updated the example